Cloudflare cache clearing recipe

suspiciousfellow opened the pull request:
Howdy doody,

We attach a cloudflare cache clearing recipie for your delight and delication.

Cheers

Cyber Duck

oanhnn commented on recipes/cloudflare.php:
```diff
@@ -0,0 +1,90 @@
+        $headers = [
+            'X-Auth-User-Service-Key' => $config['service_key']
+        ];
+    }
```
I think you should use simple code:
```
if (!empty($config['service_key'])) {
    $headers = [
        'X-Auth-User-Service-Key' => $config['service_key']
     ];
} elseif (!empty($config['email']) && !empty($config['api_key'])) {
     $headers = [
         'X-Auth-Key'   => $config['api_key'],
         'X-Auth-Email' => $config['email']
     ];
} else {
    throw new RuntimeException("Set a service key or email / api key");
}
```

oanhnn commented on recipes/cloudflare.php:
```diff
@@ -0,0 +1,90 @@
+            ),
+        ]
+    );
+
```
Please remove 2 blank lines

oanhnn commented on recipes/cloudflare.php:
```diff
@@ -0,0 +1,90 @@
+        curl_close($ch);
+
+        return $res;
+
```
Please remove a blank line

suspiciousfellow commented:
Heya,

Implemented your suggestions.

Cheers

David

oanhnn commented on recipes/cloudflare.php:
```diff
@@ -10,30 +10,25 @@
 
-
     if (empty($config['domain'])) {
         throw new RuntimeException("Set a domain");
```
can you using `\RuntimeException()` instead `RuntimeException()`

oanhnn commented on recipes/cloudflare.php:
```diff
@@ -10,30 +10,25 @@
+            'X-Auth-Email' => $config['email']
+        ];
+    } else {
+        throw new RuntimeException("Set a service key or email / api key");
```
can you using `\RuntimeException()` instead `RuntimeException()`

elfet commented on recipes/cloudflare.php:
```diff
@@ -0,0 +1,82 @@
+
+    // get the mysterious zone id from Cloud Flare
+
+    $zones = @json_decode($makeRequest(
```
Why `@`?

suspiciousfellow replied:
Force of habit when json_decoding variable that may not be a string ( json_decode([]) causes a warning), but fair point it can only be a string or false here so removed it in push coming in a second...

suspiciousfellow commented:
Changed per your comments. 

Something I noticed when creating this - the instructions for creating a pull request ask that the recipies be added in alphabetical order to the readme table, however currently the readme doesn't seem to have them in any particular order, therefore I just added this to the end. Not sure if the readme table needs updating or the instructions, but just mentioning it.

johnny-bit commented:
Update table :) apparently last PRs were not checked against this rule :)

2016-06-05 23:26 GMT+02:00 suspiciousfellow <notifications@github.com>:

> Changed per your comments.
>
> Something I noticed when creating this - the instructions for creating a
> pull request ask that the recipies be added in alphabetical order to the
> readme table, however currently the readme doesn't seem to have them in any
> particular order, therefore I just added this to the end. Not sure if the
> readme table needs updating or the instructions, but just mentioning it.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/deployphp/recipes/pull/56#issuecomment-223838741>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AKMVFaYk0aLdV1FLufNLm9wvenLDcBQUks5qIz7_gaJpZM4Iqxvt>
> .
>



-- 
Pozdrawiam,
Hubert Kowalski

suspiciousfellow commented:
@johnny-bit voila 😄

oanhnn commented:
:+1:
